<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.8.1" />
<title>pygwasvcf.variant_record_gwas_funs API documentation</title>
<meta name="description" content="Class to provide helper functions on the pysam.VariantRecord object" />
<link href='https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.0/normalize.min.css' rel='stylesheet'>
<link href='https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/8.0.0/sanitize.min.css' rel='stylesheet'>
<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" rel="stylesheet">
<style>.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>pygwasvcf.variant_record_gwas_funs</code></h1>
</header>
<section id="section-intro">
<p>Class to provide helper functions on the pysam.VariantRecord object</p>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">&#34;&#34;&#34;
Class to provide helper functions on the pysam.VariantRecord object
&#34;&#34;&#34;


# subclassing pysam.VariantRecord does not seem to work (I think this must be done in Cython)
class VariantRecordGwasFuns:
    &#34;&#34;&#34;
    Transforms -log10 P back to P value
    :param p: -log10 P value
    :return P value in the range 0,1
    &#34;&#34;&#34;

    @staticmethod
    def transform_logpval(p):
        return 10 ** -p

    &#34;&#34;&#34;
    Getter for the variant-trait association P value
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :return P value in the range 0,1
    &#34;&#34;&#34;

    @staticmethod
    def get_pval(variant_record, trait):
        p = variant_record.samples[trait][&#39;LP&#39;][0]
        if p == 0:
            return 1
        elif p == 999:
            return 0
        else:
            return VariantRecordGwasFuns.transform_logpval(p)

    &#34;&#34;&#34;
    Getter for the variant-trait association standard error value
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :return Association standard error
    &#34;&#34;&#34;

    @staticmethod
    def get_se(variant_record, trait):
        return variant_record.samples[trait][&#39;SE&#39;][0]

    &#34;&#34;&#34;
    Getter for the variant-trait beta value
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :return effect size coefficient
    &#34;&#34;&#34;

    @staticmethod
    def get_beta(variant_record, trait):
        return variant_record.samples[trait][&#39;ES&#39;][0]

    &#34;&#34;&#34;
    Getter for the variant-trait allele-frequency in the study
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :return alternative (effect) allele frequency
    &#34;&#34;&#34;

    @staticmethod
    def get_af(variant_record, trait):
        return variant_record.samples[trait][&#39;AF&#39;][0]

    &#34;&#34;&#34;
    Getter for the variant-trait variant ID
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :param create_if_missing: Create ID in the format chrom-pos-ref-alt if no ID is available
    :return Variant/marker identifier
    &#34;&#34;&#34;

    @staticmethod
    def get_id(variant_record, trait, create_if_missing=False):
        if &#34;ID&#34; in variant_record.samples[trait] and variant_record.samples[trait][&#39;ID&#39;] is not None:
            return variant_record.samples[trait][&#39;ID&#39;]
        elif create_if_missing:
            return variant_record.chrom + &#34;-&#34; + str(variant_record.pos) + &#34;-&#34; + variant_record.ref + &#34;-&#34; + \
                   variant_record.alts[0]
        else:
            raise KeyError(&#34;No ID available for this record&#34;)

    &#34;&#34;&#34;
    Getter for the variant-trait sample size used to estimate the effect
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :param metadata: If the per-variant sample size if missing then taken from global metadata (optional)
    :return Total sample size used to estimate the association effect size
    &#34;&#34;&#34;

    @staticmethod
    def get_ss(variant_record, trait, metadata=None):
        if &#39;SS&#39; in variant_record.samples[trait]:
            return variant_record.samples[trait][&#39;SS&#39;][0]
        elif metadata is not None and &#39;TotalControls&#39; in metadata[trait]:
            if &#39;TotalCases&#39; in metadata[trait]:
                return int(metadata[trait][&#39;TotalControls&#39;]) + int(metadata[trait][&#39;TotalCases&#39;])
            else:
                return int(metadata[trait][&#39;TotalControls&#39;])
        else:
            raise KeyError(&#34;No sample size available&#34;)

    &#34;&#34;&#34;
    Getter for the variant-trait number of cases size used to estimate the effect
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :param metadata: If the per-variant sample size if missing then taken from global metadata (optional)
    :return Number of cases used to estimate the association effect size
    &#34;&#34;&#34;

    @staticmethod
    def get_nc(variant_record, trait, metadata=None):
        if &#39;NC&#39; in variant_record.samples[trait]:
            return variant_record.samples[trait][&#39;NC&#39;][0]
        elif metadata is not None and &#39;TotalCases&#39; in metadata[trait]:
            return int(metadata[trait][&#39;TotalCases&#39;])
        else:
            raise KeyError(&#34;No sample size available&#34;)

    &#34;&#34;&#34;
    Check the VCF record only contains a bi-allelic change; multi-allelic changes should be on separate rows 
    :param variant_record: pysam.VariantRecord object for the VCF row
    &#34;&#34;&#34;

    @staticmethod
    def check_biallelic(variant_record):
        assert len(variant_record.alts) == 1</code></pre>
</details>
</section>
<section>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-classes">Classes</h2>
<dl>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns"><code class="flex name class">
<span>class <span class="ident">VariantRecordGwasFuns</span></span>
</code></dt>
<dd>
<div class="desc"><p>Transforms -log10 P back to P value
:param p: -log10 P value
:return P value in the range 0,1</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">class VariantRecordGwasFuns:
    &#34;&#34;&#34;
    Transforms -log10 P back to P value
    :param p: -log10 P value
    :return P value in the range 0,1
    &#34;&#34;&#34;

    @staticmethod
    def transform_logpval(p):
        return 10 ** -p

    &#34;&#34;&#34;
    Getter for the variant-trait association P value
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :return P value in the range 0,1
    &#34;&#34;&#34;

    @staticmethod
    def get_pval(variant_record, trait):
        p = variant_record.samples[trait][&#39;LP&#39;][0]
        if p == 0:
            return 1
        elif p == 999:
            return 0
        else:
            return VariantRecordGwasFuns.transform_logpval(p)

    &#34;&#34;&#34;
    Getter for the variant-trait association standard error value
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :return Association standard error
    &#34;&#34;&#34;

    @staticmethod
    def get_se(variant_record, trait):
        return variant_record.samples[trait][&#39;SE&#39;][0]

    &#34;&#34;&#34;
    Getter for the variant-trait beta value
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :return effect size coefficient
    &#34;&#34;&#34;

    @staticmethod
    def get_beta(variant_record, trait):
        return variant_record.samples[trait][&#39;ES&#39;][0]

    &#34;&#34;&#34;
    Getter for the variant-trait allele-frequency in the study
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :return alternative (effect) allele frequency
    &#34;&#34;&#34;

    @staticmethod
    def get_af(variant_record, trait):
        return variant_record.samples[trait][&#39;AF&#39;][0]

    &#34;&#34;&#34;
    Getter for the variant-trait variant ID
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :param create_if_missing: Create ID in the format chrom-pos-ref-alt if no ID is available
    :return Variant/marker identifier
    &#34;&#34;&#34;

    @staticmethod
    def get_id(variant_record, trait, create_if_missing=False):
        if &#34;ID&#34; in variant_record.samples[trait] and variant_record.samples[trait][&#39;ID&#39;] is not None:
            return variant_record.samples[trait][&#39;ID&#39;]
        elif create_if_missing:
            return variant_record.chrom + &#34;-&#34; + str(variant_record.pos) + &#34;-&#34; + variant_record.ref + &#34;-&#34; + \
                   variant_record.alts[0]
        else:
            raise KeyError(&#34;No ID available for this record&#34;)

    &#34;&#34;&#34;
    Getter for the variant-trait sample size used to estimate the effect
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :param metadata: If the per-variant sample size if missing then taken from global metadata (optional)
    :return Total sample size used to estimate the association effect size
    &#34;&#34;&#34;

    @staticmethod
    def get_ss(variant_record, trait, metadata=None):
        if &#39;SS&#39; in variant_record.samples[trait]:
            return variant_record.samples[trait][&#39;SS&#39;][0]
        elif metadata is not None and &#39;TotalControls&#39; in metadata[trait]:
            if &#39;TotalCases&#39; in metadata[trait]:
                return int(metadata[trait][&#39;TotalControls&#39;]) + int(metadata[trait][&#39;TotalCases&#39;])
            else:
                return int(metadata[trait][&#39;TotalControls&#39;])
        else:
            raise KeyError(&#34;No sample size available&#34;)

    &#34;&#34;&#34;
    Getter for the variant-trait number of cases size used to estimate the effect
    :param variant_record: pysam.VariantRecord object for the VCF row
    :param trait: Name of the trait
    :param metadata: If the per-variant sample size if missing then taken from global metadata (optional)
    :return Number of cases used to estimate the association effect size
    &#34;&#34;&#34;

    @staticmethod
    def get_nc(variant_record, trait, metadata=None):
        if &#39;NC&#39; in variant_record.samples[trait]:
            return variant_record.samples[trait][&#39;NC&#39;][0]
        elif metadata is not None and &#39;TotalCases&#39; in metadata[trait]:
            return int(metadata[trait][&#39;TotalCases&#39;])
        else:
            raise KeyError(&#34;No sample size available&#34;)

    &#34;&#34;&#34;
    Check the VCF record only contains a bi-allelic change; multi-allelic changes should be on separate rows 
    :param variant_record: pysam.VariantRecord object for the VCF row
    &#34;&#34;&#34;

    @staticmethod
    def check_biallelic(variant_record):
        assert len(variant_record.alts) == 1</code></pre>
</details>
<h3>Static methods</h3>
<dl>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.check_biallelic"><code class="name flex">
<span>def <span class="ident">check_biallelic</span></span>(<span>variant_record)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def check_biallelic(variant_record):
    assert len(variant_record.alts) == 1</code></pre>
</details>
</dd>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_af"><code class="name flex">
<span>def <span class="ident">get_af</span></span>(<span>variant_record, trait)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def get_af(variant_record, trait):
    return variant_record.samples[trait][&#39;AF&#39;][0]</code></pre>
</details>
</dd>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_beta"><code class="name flex">
<span>def <span class="ident">get_beta</span></span>(<span>variant_record, trait)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def get_beta(variant_record, trait):
    return variant_record.samples[trait][&#39;ES&#39;][0]</code></pre>
</details>
</dd>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_id"><code class="name flex">
<span>def <span class="ident">get_id</span></span>(<span>variant_record, trait, create_if_missing=False)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def get_id(variant_record, trait, create_if_missing=False):
    if &#34;ID&#34; in variant_record.samples[trait] and variant_record.samples[trait][&#39;ID&#39;] is not None:
        return variant_record.samples[trait][&#39;ID&#39;]
    elif create_if_missing:
        return variant_record.chrom + &#34;-&#34; + str(variant_record.pos) + &#34;-&#34; + variant_record.ref + &#34;-&#34; + \
               variant_record.alts[0]
    else:
        raise KeyError(&#34;No ID available for this record&#34;)</code></pre>
</details>
</dd>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_nc"><code class="name flex">
<span>def <span class="ident">get_nc</span></span>(<span>variant_record, trait, metadata=None)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def get_nc(variant_record, trait, metadata=None):
    if &#39;NC&#39; in variant_record.samples[trait]:
        return variant_record.samples[trait][&#39;NC&#39;][0]
    elif metadata is not None and &#39;TotalCases&#39; in metadata[trait]:
        return int(metadata[trait][&#39;TotalCases&#39;])
    else:
        raise KeyError(&#34;No sample size available&#34;)</code></pre>
</details>
</dd>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_pval"><code class="name flex">
<span>def <span class="ident">get_pval</span></span>(<span>variant_record, trait)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def get_pval(variant_record, trait):
    p = variant_record.samples[trait][&#39;LP&#39;][0]
    if p == 0:
        return 1
    elif p == 999:
        return 0
    else:
        return VariantRecordGwasFuns.transform_logpval(p)</code></pre>
</details>
</dd>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_se"><code class="name flex">
<span>def <span class="ident">get_se</span></span>(<span>variant_record, trait)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def get_se(variant_record, trait):
    return variant_record.samples[trait][&#39;SE&#39;][0]</code></pre>
</details>
</dd>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_ss"><code class="name flex">
<span>def <span class="ident">get_ss</span></span>(<span>variant_record, trait, metadata=None)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def get_ss(variant_record, trait, metadata=None):
    if &#39;SS&#39; in variant_record.samples[trait]:
        return variant_record.samples[trait][&#39;SS&#39;][0]
    elif metadata is not None and &#39;TotalControls&#39; in metadata[trait]:
        if &#39;TotalCases&#39; in metadata[trait]:
            return int(metadata[trait][&#39;TotalControls&#39;]) + int(metadata[trait][&#39;TotalCases&#39;])
        else:
            return int(metadata[trait][&#39;TotalControls&#39;])
    else:
        raise KeyError(&#34;No sample size available&#34;)</code></pre>
</details>
</dd>
<dt id="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.transform_logpval"><code class="name flex">
<span>def <span class="ident">transform_logpval</span></span>(<span>p)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@staticmethod
def transform_logpval(p):
    return 10 ** -p</code></pre>
</details>
</dd>
</dl>
</dd>
</dl>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="pygwasvcf" href="index.html">pygwasvcf</a></code></li>
</ul>
</li>
<li><h3><a href="#header-classes">Classes</a></h3>
<ul>
<li>
<h4><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns">VariantRecordGwasFuns</a></code></h4>
<ul class="two-column">
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.check_biallelic" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.check_biallelic">check_biallelic</a></code></li>
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_af" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_af">get_af</a></code></li>
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_beta" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_beta">get_beta</a></code></li>
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_id" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_id">get_id</a></code></li>
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_nc" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_nc">get_nc</a></code></li>
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_pval" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_pval">get_pval</a></code></li>
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_se" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_se">get_se</a></code></li>
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_ss" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.get_ss">get_ss</a></code></li>
<li><code><a title="pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.transform_logpval" href="#pygwasvcf.variant_record_gwas_funs.VariantRecordGwasFuns.transform_logpval">transform_logpval</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc"><cite>pdoc</cite> 0.8.1</a>.</p>
</footer>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad()</script>
</body>
</html>